aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,689 @@
-.. include:: /../common/stub-topic.txt
+Razor Syntax Reference
+===========================
+`Taylor Mullen <https://twitter.com/ntaylormullen>`__, and `Rick Anderson`_
 
-|stub-icon| Razor Syntax
-========================
+.. contents:: Sections
+  :local:
+  :depth: 1
 
-.. include:: /../common/stub-notice.txt
+What is Razor?
+--------------
+Razor is a markup syntax for embedding server based code into web pages. The Razor syntax consists of Razor markup, C# and HTML. Files containing Razor generally have a *.cshtml* file extension.
 
-.. _issue: https://github.com/aspnet/Docs/issues/130
+Rendering HTML
+--------------
+
+The default Razor language is HTML. Rendering HTML from Razor is no different than in an HTML file. A Razor file with the following markup:
+
+.. code-block:: html
+
+  <p>Hello World</p>
+
+Is rendered unchanged as ``<p>Hello World</p>`` by the server.
+
+Razor syntax
+----------------------
+
+Razor supports C# and uses the ``@`` symbol to transition from HTML to C#. Razor evaluates C# expressions and renders them in the HTML output. Razor can transition from HTML into C# or into Razor specific markup. When an ``@`` symbol is followed by a :ref:`Razor reserved keyword <razor-reserved-keywords-label>` it transitions into Razor specific markup, otherwise it transitions into plain C# .
+
+.. _escape-at-label:
+
+HTML containing ``@`` symbols may need to be escaped with a second ``@`` symbol. For example:
+
+.. code-block:: html
+
+ <p>@@Username</p>
+
+would render the following HTML:
+
+.. code-block:: html
+
+ <p>@Username</p>
+
+.. _razor-email-ref:
+
+HTML attributes containing email addresses don’t treat the ``@`` symbol as a transition character.
+
+ ``<a href="mailto:Support@contoso.com">Support@contoso.com</a>``
```

Comment:
Should this be an HTML code block instead of preformatted text? 

@NTaylorMullen Why isn't the `@` in the HTML *content* treated as a transition character in this case? Do we special case email addresses in both attributes and content, or just in attributes?